dashboard/app/models/script_level.rb

```diff
@@ -128,11 +128,14 @@ def final_level?
   end
 
   def next_level_or_redirect_path_for_user(user, extras_stage=nil)
-    # if we're coming from an unplugged level, it's ok to continue to unplugged
-    # level (example: if you start a sequence of assessments associated with an
-    # unplugged level you should continue on that sequence instead of skipping to
-    # next stage)
-    if valid_progression_level?(user)
+    if bubble_choice?
+      # Redirect user back to the BubbleChoice activity page.
+      level_to_follow = self
```

Comment:
please help me understand -- if we were to follow this link for a bubble choice level (not sublevel), we'd end up back on the same bubble choice level. so is there an implicit assumption that one will never be navigating to the "next" level from a bubble choice page? and this value is only getting used for sublevels of the BubbleChoice level?

Reply:
great question -- i was struggling with this area the most. the short answer is yes; long answer is addressed in #29112:

when navigating from a "parent" BubbleChoice level (rather than a sublevel), i'm manually calculating the previous/next levels in [BubbleChoice#summarize](https://github.com/code-dot-org/code-dot-org/blob/f8c6007a8f53f23c32c0c7cc0bd20c9367e21951/dashboard/app/models/levels/bubble_choice.rb#L64-L75) instead of using this method, but maybe that's a bad idea because it may create confusion in the future.

i'll take an action item to clarify this logic/intent once this PR and #29112 are merged

Reply:
That would be great -- I get why you are passing `script_level` into `summarize` and `summarize_sublevels`, but it definitely adds complexity. it seems like a win for that complexity to go away if you find a way to compute next levels a different way. maybe peek at how this is done for pages in level groups if you have not already?